diff --git a/0001-pdcurses-define-key-stub.patch b/0001-pdcurses-define-key-stub.patch
new file mode 100644
index 0000000000..e69de29bb2
diff --git a/src/nano.c b/src/nano.c
index bbfb226f34..e5a0fc3b3d 100644
--- a/src/nano.c
+++ b/src/nano.c
@@ -60,6 +60,14 @@
 # define tcgetattr(...)
 #endif

+#ifdef __PDCURSES__
+// pdcurses doesn't implement define_key.
+int define_key(const char *definition, int keycode)
+{
+  return ERR;
+}
+#endif
+
 static struct sigaction oldaction, newaction;
        /* Containers for the original and the temporary handler for SIGINT. */

