diff --git a/src/global.c b/src/global.c
index 2ff88d5b29..68c4f0f4e8 100644
--- a/src/global.c
+++ b/src/global.c
@@ -376,10 +376,10 @@
 			return -1;
 	} else if (keystring[0] == 'M') {
 		if (keystring[1] == '-' && keystring[3] == '\0') {
-			if ('A' <= keystring[2] && keystring[2] <= 'Z')
-				return (keystring[2] | 0x20);
+			if (keystring[2] > '9')
+				return (keystring[2] & 0x5F) + ALT_A - 'A';
 			else
-				return keystring[2];
+				return (keystring[2] & 0x5F) + ALT_0 - '0';
 		}
 		if (strcasecmp(keystring, "M-Space") == 0)
 			return (int)' ';
diff --git a/src/winio.c b/src/winio.c
index ecdbd9da5f..57096c10c5 100644
--- a/src/winio.c
+++ b/src/winio.c
@@ -1024,6 +1024,7 @@
 		return ERR;
 
 	if (escapes == 0) {
+		meta_key = PDC_get_key_modifiers() & PDC_KEY_MODIFIER_ALT;
 		/* Most key codes in byte range cannot be special keys. */
 		if (keycode < 0xFF && keycode != '\t' && keycode != DEL_CODE)
 			return keycode;
