diff --git a/bear/build.rs b/bear/build.rs
index 5aad005..ed1368f 100644
--- a/bear/build.rs
+++ b/bear/build.rs
@@ -29,7 +29,7 @@
 ///                   sequence ("C:\Users\..." breaks); use forward slashes
 ///                   ("C:/Users/..."), escaped backslashes ("C:\\Users\\..."),
 ///                   or a raw string literal (r"C:\Users\...")
-const DEFAULT_WRAPPER_PATH: &str = "/usr/local/libexec/bear";
+const DEFAULT_WRAPPER_PATH: &str = "../libexec/bear";
 
 /// Default preload library path
 /// Package creators: modify this entire path to change preload library location
